### DISCUSSION TITLE
Roles in recipe

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#8691] [Q&A] [answered]
How can i remove default roles (Editor, Author etc) by recipe?

**Comment by vitalybrandes** [comment, marked as answer]
Or any other way.....

**Reply by Skrypt** [reply]
I don't recommend removing the default roles as they are used in different admin modules for security.

**Reply by vitalybrandes** [reply]
@Skrypt  This affect security issues even if i am not using this roles at all?

**Reply by Skrypt** [reply]
Orchard Core uses some of them. It will break also some of our themes recipes. It's better to keep them than removing them. For example the Anonymous role is important and the Administrator role too. If you are talking about EditorRole I'd need to investigate to see if we hard coded it in some of our modules but I would not be surprised. So, yes, from memory I think it's used in the OrchardCore.Contents module for example. So, it is not a good idea to remove them. Generally, all you need is to add custom ones for your usage.

**Reply by vitalybrandes** [reply]
Thank you!
I will leave them as is.for now.
Is there any way i can anyhow to remove then, to test what will happen ☺️?
I agree about admin, anonymous and registered, they are must, regarding other  🤔, i think they are more optional.

**Reply by Skrypt** [reply]
These roles are merged from all the Permissions.cs files we have. Look at AddDefaultRolesForFeatureAsync in RoleUpdater.cs.
That's my only guess for now. Not sure it's a good idea still 😉

**Reply by vitalybrandes** [reply]
Thanks!!!!

**Reply by Skrypt** [reply]
I just tried locally on a running website. It is possible to remove all the roles except Authenticated and Anonymous. I'm not sure yet how we could remove all the other ones from recipes, but it technically works. All you need is a "superuser" account to access the admin dashboard. Though, like I said, modules are predefining permissions for theses roles which you will lose in the process. Maybe, they are not necessary for you and recipes should work fine from themes even if they are removed. Though, we don't have any recipe to remove roles. Maybe what you need is a custom module that will remove these specific roles.

**Reply by vitalybrandes** [reply]
Thank you
